src/NHibernate/Cfg/Loquacious/CacheConfiguration.cs

```diff
@@ -39,7 +40,13 @@ public int DefaultExpiration
 			cfg.SetProperty(Environment.CacheProvider, typeof(TProvider).AssemblyQualifiedName);
 		}
 
-		public void QueryCache<TFactory>() where TFactory : IQueryCacheFactory
+		[Obsolete("This method is invalid and should not be used. Use QueryCacheFactory method instead.", true)]
+		public void QueryCache<TFactory>() where TFactory : IQueryCache
+		{
+			throw new InvalidOperationException("This method is invalid and should not be used. Use QueryCacheFactory method instead.");
+		}
+
+		public void QueryCacheFactory<TFactory>() where TFactory : IQueryCacheFactory
```

Comment:
The general wording for method setting factory seems to include `Factory` as a suffix to the method name. So here we can dodge the overload trouble by renaming the method.